src/effects/effectslot.cpp

```diff
@@ -161,11 +160,11 @@ void EffectSlot::loadEffect(EffectPointer pEffect) {
             addEffectButtonParameterSlot();
         }
 
-        foreach (EffectParameterSlotPointer pParameter, m_parameters) {
+        for (EffectParameterSlotPointer pParameter : m_parameters) {
             pParameter->loadEffect(pEffect);
         }
 
-        foreach (EffectButtonParameterSlotPointer pParameter, m_buttonParameters) {
+        for (EffectButtonParameterSlotPointer pParameter : m_buttonParameters) {
```

Comment:
Please use ```const auto&```. to avoid a deep copy. 

As a rule of thump, you can always use use ```const auto&```, and if the compiler complains about ```const``` use the universal reference ```auto&&```. This allows the complier to use optimal access and avoids implicit conversion.

Exceptions are raw pointers and 32 bit values where the reference stuff is not required.   
If you need a non const loop for implicit shared QT containers, use the Qt foreach macro. 

Byt the way: 
const in this context means that the QSharedPointer is const. The pointer carried by a shared pointer is always non const.